src/Authentication/JWTAuthenticator.php

```diff
@@ -155,71 +271,104 @@ public function generateToken(Member $member)
     /**
      * @param string $token
      * @param HTTPRequest $request
-     * @param ValidationResult $result
-     * @return null|Member
+     * @return array Array with JWTRecord and int status (STATUS_*)
      * @throws BadMethodCallException
      */
-    private function validateToken($token, $request, &$result)
+    public function validateToken(string $token, HTTPrequest $request): array
```

Comment:
Actual valid issue from CodeClimate: It's quite a long method:
https://codeclimate.com/github/Firesphere/silverstripe-graphql-jwt/src/Authentication/JWTAuthenticator.php/source?from_sha=2a45ae2a5722e3685f0c5db06f46dbd3b6eb80d2&to_sha=8fd6b877d78de4d520a9eb566a757e017bbb8ea5
And has a lot of return statements.

Could you reduce the returns (maybe not an early return each time?) and maybe refactor the ValidationData part in to a separate getter?

Reply:
I've broken this into a bunch of methods. There are still a few early returns, but a few fewer than before.